iothub/device/src/Authentication/DeviceAuthenticationWithTpm.cs

```diff
@@ -68,7 +69,16 @@ protected override string Sign(string requestString, string key)
                 Debug.Assert(key == null);
 
                 byte[] encodedBytes = Encoding.UTF8.GetBytes(requestString);
-                byte[] hmac = _authenticationProvider.Sign(encodedBytes);
+                byte[] hmac = Array.Empty<byte>();
+                try
+                {
+                    hmac = _authenticationProvider.Sign(encodedBytes);
+                }
+                catch (Exception ex) when (ex is TssException || ex is TpmException)
```

Comment:
I believe one of these exceptions inherits from the other, so your check can be against only the base type.